src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,59 @@
+# I haven't checked ``downsample`` parameter as it is always performed after every block.
+function basicblock(planes::Int)
+  expansion = 1;
+  if groups != 1 || base_width != 64
+    error('basicblock only supports groups=1 and base_width=64');
+  end
+  if dilation > 1
+    error("Dilation > 1 not supported in BasicBlock");
+  end
+  basic =  Chain(
+        Conv((3, 3), planes ÷ 2 => planes, stride = (1,1), pad = (1,1), dilation = 1),
+        BatchNorm(planes, λ = relu),
+        Conv((3, 3), planes => planes, stride = (1,1), pad = (1,1), dilation = 1),
+        BatchNorm(planes),
+        Sequential(Conv((1, 1), planes ÷ 2 => planes * expansion, stride=(1, 1)),
+               BatchNorm(planes * expansion, λ=relu)
+        )
+      )
+  return basic;
+end
+# I haven't checked ``downsample`` parameter as it is always performed after every block.
+function bottleneck(planes::Int, )
+  expansion = 4;
+  base_width = 64;
+  groups = 1;
+  width = int(planes * (base_width / 64.)) * groups
+  residual =  Chain(
+        Conv((1, 1), planes=>width, stride=(2,2)),
+        BatchNorm(planes, λ=relu),
+        Conv((3, 3), width=>width, stride=(2, 2), pad=(1, 1), groups=1, dilation=1),
+        BatchNorm(planes, λ=relu),
+        Conv((1, 1), width=>planes * expansion, stride=(2,2)),
+        BatchNorm(planes * expansion, λ=relu),
+        Sequential(Conv((1, 1), width=>planes * expansion, stride=(2, 2)),
+               BatchNorm(planes * expansion, λ=relu)
+        )
+      )
+  return residual;
+end 
```

Comment:
The ResNet implementation in Metalhead is designed for pre-trained weights from PyTorch. I would prefer to switch this implementation to use [SkipConnection](https://fluxml.ai/Flux.jl/stable/models/layers/#Flux.SkipConnection) layer in Flux.

Reply:
using SkipConnection layer we can wrap up the whole code in one single function rather than having two function calls. Only different inputs' function will be required and a main resnet().